File: 0066-plus-one/0066-plus-one.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int size = digits.size();
        int count = 0;
        if(digits[size-1]==9)
        {
            for(int i =1; i<=size; i++)
            {
                if(digits[size-i]==9)
                {
                    digits[size-i]=0;
                    count = count + 1;
                }
                else
                {
                    digits[size-i]=digits[size-i]+1;
                    break;
                }
            }
        }
        else
        {
           digits[size-1]=digits[size-1]+1; 
        }
        if(count == size)
        {
            vector<int> digit;
            digit.push_back(1);
            for(int i=0; i<size;i++)
            {
                digit.push_back(0);
            }
            return digit;
        }
        else
        {
            return digits;
        }
    }
};

File: 0066-plus-one/0066-plus-one.py
class Solution(object):
    def plusOne(self, arr):
        size = len(arr)
        for i in range(size):
            if(arr[size-1-i]==9):
                arr[size-1-i] = 0
                if i == size-1:
                    arr.insert(0,1)
                    return arr
                else:
                    pass
            else:
                arr[size-1-i] = arr[size-1-i] + 1
                print(arr)
                return arr